build_sdk_container_image

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -eu
+
+cd $(dirname "$0")
+source sdk_lib/sdk_container_common.sh
+
+tarball=""
+arch="amd64"
+checkout="true"
+dev="false"
+publish="false"
+
+version="$(get_version_from_versionfile)"
+# --
+
+usage() {
+    echo "  $0 - Create SDK container image from an SDK tarball"
+    echo "       This script will set up a new SDK container from a tarball."
+    echo "       The resulting container comes in 3 flavours:"
+    echo "        1.    flatcar-sdk-all - includes both ARM64 and AMD64 support"
+    echo "        2.+3. flatcar-sdk-(amd64|arm64) - only includes suuport for one target."
+    echo "  Usage:"
+    echo "  $0 [-v <version> -a <arch> | -f <tarball>] [-d ] [-c] [-p]"
+    echo
+    echo "      -v <version> - SDK version to containerise. Default: '$version'"
+    echo "                     (from '$sdk_container_common_versionfile')"
+    echo "      -a <arch>    - Architecture the SDK runs on; either 'amd64' or 'arm64'."
+    echo "                     NOTE this is different from the target architecture"
+    echo "                     (the SDK can produce binaries for). Default: '$arch'"
+    echo "      -f <tarball> - Build from a local tarball instead of downloading."
+    echo "                     This overrides -v/-a. The tarball must follow the format"
+    echo "                     flatcar-sdk-(amd64|arm64)-<version>.tar.bz2."
+    echo "      -c           - Use scripts and ebuild repos as-is, do not check out version tag."
+    echo "      -d             Indicates the container to be built is based on a development/nightly"
+    echo "                     SDK version, not a released SDK"
+    echo "      -p             Publish containers to '$sdk_container_common_registry' after successful build."
+    echo "                     requires docker to be authenticated to the registry."
+    echo "      -h             Print help"
+    echo
+}
+
+# --
+
+while [ 0 -lt $# ] ; do
+    case "$1" in
+    -h) usage; exit 0;;
+    -v) version="$2"; shift; shift;;
+    -a) arch="$2";    shift; shift;;
+    -f) tarball="$2"; shift; shift;;
+    -c) checkout="false";    shift;;
+    -d) dev="true";          shift;;
+    -p) publish="true";      shift;;
+    *)  usage; exit 1;;
+    esac
+done
+
+# --
+
+if [ -z "$version" ] ; then
+    echo "ERROR: Flatcar SDK version to build container for cannot be autodetected and was not provided."
+    exit 1
+fi
+
+yell "\n######\n###### Building SDK container for version $version"
+
+if $checkout; then
```

Comment:
I think this should always ensure that the submodule commits are matching because this is the tricky part of using submodules. Maybe this will require to do a local commit in flatcar-scripts but I think it's worth to be as strict as possible?

Reply:
I'm currently reworking version handling so the git tree version is authoritative. The version file will always be generated.
If the commit has a tag, then I'll use the tag (`git tag --points-at HEAD`). If there's no tag then I'll use `git describe` output.